src/praisonai/praisonai/persistence/conversation/_sqlite_base.py

```
@@ -0,0 +1,356 @@
+            )
+            rows = cur.fetchall()
+        return [self._row_to_message(row) for row in rows]
+
+    def delete_messages(self, session_id: str, message_ids: Optional[List[str]] = None) -> int:
+        """Delete messages. If message_ids is None, delete all messages in session."""
+        with self._connection() as conn:
+            if message_ids:
+                placeholders = ",".join(["?"] * len(message_ids))
+                cur = conn.execute(
+                    f"DELETE FROM {self.messages_table}"
+                    f" WHERE session_id = ? AND id IN ({placeholders})",
```

Ruff (0.15.18)

[error] Possible SQL injection vector through string-based query construction

(S608)